File: backend/app/schemas/invoice.py
"""Pydantic v2 schemas for Invoice endpoints."""
from datetime import date, datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class InvoiceLineItemOut(BaseModel):
    id: UUID
    invoice_id: UUID
    description: str | None = None
    quantity: Decimal
    unit_price: Decimal
    total: Decimal
    support_item_number: str | None = None

    model_config = ConfigDict(from_attributes=True)


class InvoiceOut(BaseModel):
    id: UUID
    participant_id: UUID | None = None
    provider_id: UUID | None = None
    invoice_number: str | None = None
    invoice_date: date | None = None
    due_date: date | None = None
    total_amount: Decimal
    gst_amount: Decimal
    status: str
    ocr_confidence: float | None = None
    gcs_pdf_path: str | None = None
    gcs_json_path: str | None = None
    line_items: list[InvoiceLineItemOut] = []
    created_at: datetime
    participant_approved: bool | None = None
    participant_approved_at: datetime | None = None
    participant_query_message: str | None = None

    model_config = ConfigDict(from_attributes=True)


class InvoiceListOut(BaseModel):
    items: list[InvoiceOut]
    total: int
    page: int
    page_size: int

File: frontend/lib/types/invoice.ts
export type InvoiceStatus =
  | 'queued'
  | 'flagged'
  | 'pending_approval'
  | 'info_requested'
  | 'approved'
  | 'rejected'
  | 'paid'

export interface Invoice {
  id: string
  participant_id: string | null
  provider_id: string | null
  invoice_number: string | null
  invoice_date: string | null
  due_date: string | null
  total_amount: string
  gst_amount: string
  status: InvoiceStatus
  ocr_confidence: number | null
  gcs_pdf_path: string
  line_items: InvoiceLineItem[]
  validation_results: ValidationResult[] | null
  reviewed_by: string | null
  reviewed_at: string | null
  created_at: string
  participant_approved: boolean | null
  participant_approved_at: string | null
  participant_query_message: string | null
}

export interface InvoiceLineItem {
  id: string
  support_item_number: string | null
  description: string | null
  unit_price: string
  quantity: string
  total: string
}

export interface ValidationResult {
  rule_name: string
  passed: boolean
  message: string
  severity: 'error' | 'warning'
}

export interface InvoiceListResponse {
  items: Invoice[]
  total: number
  page: number
  page_size: number
}

export interface ValidationReport {
  invoice_id: string
  validated_at: string
  results: ValidationResult[]
  overall_passed: boolean
}
